src/eradiate/scenes/atmosphere/_molecular_atmosphere.py

```diff
@@ -98,21 +105,21 @@ def _switch_validator(self, attribute, value):
                 "and 'has_scattering' must be True"
             )
 
-    absorption_data_sets: dict[str, str] | None = documented(
+    absorption_dataset: xr.Dataset | None = documented(
         attrs.field(
             default=None,
-            converter=attrs.converters.optional(dict),
-            validator=attrs.validators.optional(attrs.validators.instance_of(dict)),
+            converter=attrs.converters.optional(
+                converters.to_dataset(load_from_id=None)
+            ),
+            validator=attrs.validators.optional(
+                attrs.validators.instance_of(xr.Dataset)
+            ),
         ),
-        doc="Mapping of species and absorption data set files paths. If "
-        "``None``, the default absorption data sets are used to compute "
-        "the absorption coefficient. If not ``None``, the absorption data "
-        "set files whose paths are provided in the mapping will be used to "
-        "compute the absorption coefficient. If the mapping does not "
-        "include all species from the atmospheric "
-        "thermophysical profile, the default data sets will be used to "
-        "compute the absorption coefficient of the corresponding species.",
-        type="dict",
+        doc="Absorption coefficient dataset. If ``None``, the absorption "
+        "coefficient is set to zero.",
+        type=":class:`~xarray.Dataset`",
+        init_type="PathLike or :class:`~xarray.Dataset`",
```

Comment:
Use Sphinx aliases for clarity in the init type field:
```suggestion
        init_type="path-like or Dataset`",
```